### PR TITLE
ddl: migrate test-infra to testify for primary_key_handle_test.go

### LINKED ISSUE #32843: migrate test-infra to testify for `ddl/primary_key_handle_test.go`

### PR DESCRIPTION
**Pull request by tisonkun**
### What problem does this PR solve?

Issue Number: close #32843 

```release-note
None
```

### CODE REVIEW

File: ddl/db_integration_test.go [hunk 1]
```diff
@@ -2220,6 +2220,10 @@ func TestDropAutoIncrementIndex(t *testing.T) {
 	tk.MustExec("create database if not exists test")
 	tk.MustExec("use test")
 
+	tk.MustExec("drop table if exists t1")
```
Comment (tisonkun) [inline ddl/db_integration_test.go:2223]: Merge `TestDropAutoIncrementIndex` in `primary_key_handle_test.go`.

**Review by hawkingrei — APPROVED** [review]

**Review by crazycs520 — APPROVED** [review]

**Pull request merged**